plugins/module_utils/openssh/backends/common.py

```diff
@@ -18,25 +18,289 @@
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
+import abc
 import os
+import stat
 
+from ansible.module_utils import six
 
-def restore_on_failure(f):
-    def backup_and_restore(module, path, *args, **kwargs):
-        backup_file = module.backup_local(path) if os.path.exists(path) else None
+from ansible_collections.community.crypto.plugins.module_utils.openssh.utils import (
+    parse_openssh_version,
+)
+
+
+def _restore_on_failure(f):
+    def backup_and_restore(self, sources_and_destinations, *args, **kwargs):
+        backups = [(d, self.module.backup_local(d) if os.path.exists(d) else None) for s, d in sources_and_destinations]
 
         try:
-            f(module, path, *args, **kwargs)
+            f(self, sources_and_destinations, *args, **kwargs)
         except Exception:
-            if backup_file is not None:
-                module.atomic_move(backup_file, path)
+            for destination, backup in backups:
+                if backup is not None:
```

Comment:
With the above change, this (and the similar line below) wouldn't be needed anymore.